addons/binding/org.openhab.binding.sinope/src/main/java/org/openhab/binding/sinope/handler/SinopeThermostatHandler.java

```diff
@@ -0,0 +1,279 @@
+import ca.tulip.sinope.core.internal.SinopeDataAnswer;
+import ca.tulip.sinope.util.ByteUtil;
+
+/**
+ * The {@link SinopeThermostatHandler} is responsible for handling commands, which are
+ * sent to one of the channels.
+ *
+ * @author Pascal Larin - Initial contribution
+ */
+public class SinopeThermostatHandler extends BaseThingHandler implements ThingHandler {
+
+    private Logger logger = LoggerFactory.getLogger(SinopeThermostatHandler.class);
+
+    private SinopeGatewayHandler gatewayHandler;
+
+    public SinopeThermostatHandler(Thing thing) {
+        super(thing);
+    }
+
+    @Override
+    public void handleCommand(ChannelUID channelUID, Command command) {
+        Channel channel = getThing().getChannel(channelUID.getId());
+        SinopeGatewayHandler gateway = getSinopeGatewayHandler();
+        gateway.stopPoll();
+        try {
+            if (channel != null && SinopeBindingConstants.CHANNEL_SETTEMP.equals(channelUID.getId())
+                    && command instanceof DecimalType) {
+                setSetpointTemp(this, ((DecimalType) command).floatValue());
+            }
+
+            if (channel != null && SinopeBindingConstants.CHANNEL_SETMODE.equals(channelUID.getId())
+                    && command instanceof StringType) {
+                setSetpointMode(this, Integer.parseInt(((StringType) command).toString()));
+            }
+        } catch (IOException e) {
+            logger.error("Cannot handle command for channel {} because of {}", channelUID.getId(),
+                    e.getLocalizedMessage());
+            gateway.setCommunicationError(true);
+
+        }
+        gateway.schedulePoll();
+    }
+
+    public void setSetpointTemp(SinopeThermostatHandler sinopeThermostatHandler, float temp)
+            throws UnknownHostException, IOException {
+        int newTemp = (int) (temp * 100.0);
+        byte[] deviceId = SinopeConfig.convert(sinopeThermostatHandler.getDeviceId());
+        if (deviceId == null) {
+            logger.error("Invalid Device id: {}", deviceId);
+            updateStatus(ThingStatus.UNINITIALIZED);
+            return;
+        }
+
+        SinopeGatewayHandler gateway = getSinopeGatewayHandler();
+        if (gateway.connectToBridge()) {
+
+            logger.debug("Connected to bridge");
+
+            SinopeDataWriteRequest req = new SinopeDataWriteRequest(gateway.newSeq(), deviceId,
+                    new SinopeSetPointTempData());
+            ((SinopeSetPointTempData) req.getAppData()).setSetPointTemp(newTemp);
+
+            SinopeDataAnswer answ = (SinopeDataAnswer) gateway.execute(req);
+
+            if (answ.getStatus() == 0) {
+                logger.debug("Setpoint temp is now: {} C", newTemp);
+            } else {
+                logger.debug("Cannot Setpoint temp, status: {}", answ.getStatus());
+            }
+        } else {
+            logger.error("Could not connect to bridge to update Setpoint Temp");
```

Comment:
How about setting the thing status to OFFLINE here with ThingStatusDetail.COMMUNICATION_ERROR?

Reply:
There was a "instead" missing in @triller-telekom comment - you must not log an error here.